### PR TITLE
[Stdlib] Add missing comparison operators for StringSlice with String

### PR DESCRIPTION
**Pull request by msaelices**
## Summary
- Implement missing comparison operators (`<`, `<=`, `>`, `>=`) for StringSlice compared with String
- Add comprehensive test coverage for all comparison operations including edge cases
- Enables natural comparison syntax like `string_slice < "abc"`

## Examples

It's useful for comparison with string or string literals in functions that receives string slices coming from String to avoid allocations:

```mojo
fn is_digit(StringSlice s) -> Int:
    return s >= "0" and s <= "9"
```

### CODE REVIEW

File: mojo/stdlib/stdlib/collections/string/string_slice.mojo [hunk 1]
```diff
@@ -968,6 +968,55 @@ struct StringSlice[mut: Bool, //, origin: Origin[mut]](
...
+    fn __le__(self, rhs: String) -> Bool:
+        """Define whether this String slice is less than or equal to the RHS.
+
+        Args:
+            rhs: The other String to compare against.
+
+        Returns:
+            True if this String slice is less than or equal to the RHS String.
+        """
+        return not (rhs.as_string_slice() < self)
```
Comment (laszlokindrat) [inline mojo/stdlib/stdlib/collections/string/string_slice.mojo]: Here and below: instead of implementing these in terms of `<`, could we just implement in terms of the corresponding implementation that takes StringSlice?
```suggestion
        return self <= rhs.as_string_slice()
```

Comment (msaelices) [reply mojo/stdlib/stdlib/collections/string/string_slice.mojo]: Sure. It's not as easy as it seems as there were a couple of missing comparisons operators to implement. 

Fixed here: https://github.com/msaelices/mojo/commit/e85ccd76b466e82c29b98fa27ac24cca11a3ef33

Comment (msaelices) [reply mojo/stdlib/stdlib/collections/string/string_slice.mojo]: @laszlokindrat could you PTAL?

**Review by laszlokindrat — APPROVED** [review]
Thank you!

**Comment by laszlokindrat** [comment]
!sync
